lib/lobby.js

```diff
@@ -0,0 +1,55 @@
+const { sleep } = require("./helpers");
+
+class Lobby {
+  constructor(waitingPlayers = [], games = new Set()) {
+    this.waitingPlayers = waitingPlayers;
+    this.games = games;
+  }
+  addGame(game) {
+    this.games.add(game);
+  }
+  addAsWaitingPlayer(player) {
+    if (!this.waitingPlayers.includes(player)) {
+      player.log("Adding you to the lobby");
+      this.waitingPlayers.push(player);
+    }
+  }
+  getGameFromPlayer(player) {
+    for (let game of this.games) {
+      if (Object.values(game.players).includes(player)) {
+        return game;
+      }
+    }
+  }
+  matchPlayers(player) {
+    if (this.waitingPlayers.length > 1) {
+      const player1 = this.waitingPlayers.pop();
+      const player2 = this.waitingPlayers.pop();
+      return [player1, player2];
+    } else {
+      return false;
```

Comment:
```suggestion
      return null;
```

generally I would only return `false` one arm of a function if the other arm returns `true`. If I have a function that returns a complex object, the "falsy" arm would usually return `null`. This is mostly done to make it more similar to C/C++/Java/Go, where it is not possible to have a return type of `SomeObject | boolean`. Only `| null` is allowed there.